runtime/gc_modron_startup/mmparse.cpp

```diff
@@ -1369,7 +1369,14 @@ scan_udata_memory_size_helper(J9JavaVM *javaVM, char **cursor, UDATA *value, con
 		return false;
 	}
 	
-	if(try_scan(cursor, "G") || try_scan(cursor, "g")) {
+	if(try_scan(cursor, "T") || try_scan(cursor, "t")) {
+		if (*value <= (((UDATA)-1) >> 40)) {
+			*value <<= 40;
+		} else {
```

Comment:
I expect the 32-bit windows compile will fail here (shifting by amounts greater than the width of `UDATA`).

Reply:
@avermeer given Keith's comment, let's use an #ifdef to exclude the bit shift code on 32-bit. Even if it compiles on Windows 32-bit, the code does successfully compile for 32-bit on Java 8 for most platforms and we'd like to keep it that way, although they aren't officially supported or testable at the OpenJ9 project.

Reply:
> I expect the 32-bit windows compile will fail here (shifting by amounts greater than the width of `UDATA`).

As I remember it does mod32 (or circular) shift, so it is shift by 8 most likely

Reply:
I believe you can do something like this:
``` 
       if(try_scan(cursor, "T") || try_scan(cursor, "t")) {
#if defined(J9VM_ENV_DATA64)
		if (*value <= (((UDATA)-1) >> 40)) {
 			*value <<= 40;
 		} else
#endif /* defined(J9VM_ENV_DATA64) */ 
               {
			j9nls_printf(PORTLIB, J9NLS_ERROR, J9NLS_GC_OPTIONS_VALUE_OVERFLOWED, argName);
 			return false;
 		}
        } else if(try_scan(cursor, "G") || try_scan(cursor, "g")) {
```

Reply:
Yes, it is technically correct to support zero (`0T`) in the parsing functions.